File: spec/java/src/io/kaitai/struct/spec/TestEnumInvalid.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.EnumInvalid;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestEnumInvalid extends CommonSpec {

    @Test
    public void testEnumInvalid() throws Exception {
        EnumInvalid r = EnumInvalid.fromFile(SRC_DIR + "term_strz.bin");

        assertEquals(r.pet1(), EnumInvalid.Animal.Known.DOG);
        assertEquals(r.pet2(), new EnumInvalid.Animal.Unknown(111));
    }
}

File: spec/lua/test_enum_invalid.lua
local luaunit = require("luaunit")

require("enum_invalid")

TestEnumInvalid = {}

function TestEnumInvalid:test_enum_invalid()
    local r = EnumInvalid:from_file("src/term_strz.bin")

    luaunit.assertEquals(r.pet_1, EnumInvalid.Animal.dog)
    luaunit.assertNil(r.pet_2)
end
